### PR TITLE
TracksToParticles: propagate MC associations from Tracks

### PR DESCRIPTION
**Pull request by veprbl**
### Briefly, what does this PR introduce?
The https://github.com/eic/EICrecon/pull/1564 had implemented meaningful associations based on hits used. That information became available for Tracks, but end users use ReconstructedParticles, and that's what we'd like to have implemented in this PR.

Closes: #1356

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No

There will be new warnings like
```
[PFRICH:RICHEndcapNLUTPID] [warning] Found a duplicate association for ReconstructedParticle at index 18
[PFRICH:RICHEndcapNLUTPID] [warning] The previous MCParticle was at 149 and the duplicate is at 200
```
those are mostly harmless and are to be addressed separately.

### Does this PR change default behavior?
Yes

### CODE REVIEW

File: src/algorithms/tracking/TracksToParticles.cc [hunk 1]
```diff
@@ -1,26 +1,20 @@
 // SPDX-License-Identifier: LGPL-3.0-or-later
 // Copyright (C) 2022 - 2024, Sylvester Joosten, Wouter Deconinck, Dmitry Romanov, Christopher Dilks, Dmitry Kalinkin
 
-#include <algorithms/logger.h>
 #include <edm4eic/TrackParametersCollection.h>
 #include <edm4eic/TrajectoryCollection.h>
-#include <edm4hep/ParticleIDCollection.h>
+#include <edm4hep/MCParticle.h>
```
Comment (wdconinc) [inline src/algorithms/tracking/TracksToParticles.cc:6]:
```suggestion
#include <edm4hep/MCParticleCollection.h>
```

File: src/algorithms/tracking/TracksToParticles.cc [hunk 2]
```diff
@@ -46,151 +35,44 @@ namespace eicrecon {
...
+                    part_assoc.setSimID(part_assoc.getSim().getObjectID().index);
+                    part_assoc.setWeight(track_assoc.getWeight());
+
+                    if (max_weight < track_assoc.getWeight()) {
+                        max_weight = track_assoc.getWeight();
+                        edm4hep::Vector3f referencePoint = {
+                            static_cast<float>(track_assoc.getSim().getVertex().x),
+                            static_cast<float>(track_assoc.getSim().getVertex().y),
+                            static_cast<float>(track_assoc.getSim().getVertex().z)}; // @TODO: not sure if vertex/reference point makes sense here
+                        rec_part.setReferencePoint(referencePoint);
```
Comment (wdconinc) [inline src/algorithms/tracking/TracksToParticles.cc:71]: What's the problem with track.position? https://github.com/eic/EDM4eic/blob/main/edm4eic.yaml#L427

**Review by wdconinc — DISMISSED** [review]
Looks fine to me.

File: src/algorithms/tracking/TracksToParticles.cc (hunk 2 above)
Comment (veprbl) [reply src/algorithms/tracking/TracksToParticles.cc:71]: Nothing is wrong. I assume, this position will come from vertex in the near future.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**